File: 백준/Gold/17951. 흩날리는 시험지 속에서 내 평점이 느껴진거야/흩날리는 시험지 속에서 내 평점이 느껴진거야.py
#흩날리는 시험지 속에서 내 평점이 느껴진거야
def check(num):
    cnt = 0
    tmp = 0
    for i in d:
        tmp += i
        if tmp >= num:
            tmp = 0
            cnt += 1

    return cnt >= k

n,k = map(int,input().split())
d = list(map(int,input().split()))

s,e = 0,int(20e5)+1
while s+1<e:
    mid = (s+e)//2

    if check(mid):
        s = mid

    else:
        e = mid

print(s)

File: 백준/Gold/17951. 흩날리는 시험지 속에서 내 평점이 느껴진거야/흩날리는 시험지 속에서 내 평점이 느껴진거야.cc
#include <iostream>
#include <vector>

using namespace std;

int i,tmp;
int n,k;
vector<int> v;

bool check(int num) {
    int cnt = 0;
    int tmp = 0;
    for (i=0; i<v.size(); i++) {
        tmp += v[i];
        if (tmp >= num) {
            tmp = 0;
            cnt++;
        }
    }
    return (cnt >= k);
}

int main() {
    int s,e,mid;

    cin >> n >> k;
    for (i=0; i<n; i++) {
        cin >> tmp;
        v.push_back(tmp);
    }

    s = 0;
    e = 20e5+1;
    while (s+1<e) {
        mid = (s+e)/2;

        if (check(mid)) {
            s = mid;
        }
        else { 
            e = mid;
        }
    }
    cout << s << '\n';

    return 0;
}